AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/helper/InjectHelper.java

```diff
@@ -0,0 +1,152 @@
+
+	private final ValidatorHelper validatorHelper;
+	private final HasMethodInjection handler;
+
+	public InjectHelper(ValidatorHelper validatorHelper, HasMethodInjection handler) {
+		this.validatorHelper = validatorHelper;
+		this.handler = handler;
+	}
+
+	public void validate(Class<? extends Annotation> expectedAnnotation, Element element, ElementValidation valid) {
+		Element enclosingElement = element.getEnclosingElement();
+		if (element instanceof VariableElement && enclosingElement instanceof ExecutableElement) {
+			validatorHelper.enclosingElementHasEnhancedComponentAnnotation(enclosingElement, valid);
+			validatorHelper.param.annotatedWith(expectedAnnotation).multiple().validate((ExecutableElement) enclosingElement, valid);
+		} else {
+			validatorHelper.enclosingElementHasEnhancedComponentAnnotation(element, valid);
+		}
+	}
+
+	public void process(Element element, EComponentHolder holder) {
+		if (element instanceof ExecutableElement) {
+			processMethod(element, holder);
+		} else {
+			Element enclosingElement = element.getEnclosingElement();
+			if (enclosingElement instanceof ExecutableElement) {
+				processParam(element, holder);
+			} else {
+				processField(element, holder);
+			}
+		}
+	}
+
+	private void processParam(Element element, EComponentHolder holder) {
+		ExecutableElement method = (ExecutableElement) element.getEnclosingElement();
+		List<? extends VariableElement> parameters = method.getParameters();
+		List<ParamHelper> paramHelper = methodParameterMap.get(method);
+		int paramCount = parameters.size();
+
+		if (paramHelper == null) {
+			paramHelper = new ArrayList<>();
+			methodParameterMap.put(method, paramHelper);
+		}
+
+		for (int paramIndex = 0; paramIndex < paramCount; paramIndex++) {
+			VariableElement param = parameters.get(paramIndex);
+			if (param.equals(element)) {
+				JExpression getInstance = handler.getInstanceInvocation(param, holder, param);
+				paramHelper.add(new ParamHelper(getInstance, paramIndex));
+			}
+		}
+
+		if (paramHelper.size() == paramCount) {
+			String methodName = method.getSimpleName().toString();
+
+			Collections.sort(paramHelper, new Comparator<ParamHelper>() {
+
+				@Override
+				public int compare(ParamHelper o1, ParamHelper o2) {
+					return o1.argumentOrder - o2.argumentOrder;
+				}
+			});
+
+			JBlock block = holder.getInitBody();
+			JInvocation invocation = block.invoke(methodName);
+			for (ParamHelper paramHelper2 : paramHelper) {
+				invocation.arg(paramHelper2.beanInstance);
+			}
+
+			methodParameterMap.remove(method);
+		}
+	}
+
+	private void processField(Element element, EComponentHolder holder) {
+		String fieldName = element.getSimpleName().toString();
+		JFieldRef beanField = ref(fieldName);
+		JBlock block = holder.getInitBody();
+
+		// TODO check if always required...
+		boolean hasNonConfigurationInstanceAnnotation = element.getAnnotation(NonConfigurationInstance.class) != null;
```

Comment:
I see. So this checks if the field is annotated with both `@NonConfigurationInstance` and `@Bean`, the bean should not be reinjected after configuration changes. I think this only applies to beans, because views, resources etc should not be annotated with `@NonConfigurationInstance`, as the javadoc says.